opencensus/common/monitored_resource_util/gcp_metadata_config.py

```diff
@@ -0,0 +1,158 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from opencensus.common.http_handler import get_request
+import os
+
+_GCP_METADATA_URI = 'http://metadata/computeMetadata/v1/'
+_GCP_METADATA_URI_HEADER = {'Metadata-Flavor': 'Google'}
+
+# GCE common attributes
+# See: https://cloud.google.com/appengine/docs/flexible/python/runtime#
+#      environment_variables
+_GCE_ATTRIBUTES = {
+    # ProjectID is the identifier of the GCP project associated with this
+    # resource, such as "my-project".
+    'project_id': 'project/project-id',
+
+    # instance_id is the numeric VM instance identifier assigned by
+    # Compute Engine.
+    'instance_id': 'instance/id',
+
+    # zone is the Compute Engine zone in which the VM is running.
+    'zone': 'instance/zone'
+}
+
+_GKE_ATTRIBUTES = {
+    # ProjectID is the identifier of the GCP project associated with this
+    # resource, such as "my-project".
+    'project_id': 'project/project-id',
+
+    # instance_id is the numeric VM instance identifier assigned by
+    # Compute Engine.
+    'instance_id': 'instance/id',
+
+    # zone is the Compute Engine zone in which the VM is running.
+    'zone': 'instance/zone',
+
+    # cluster_name is the name for the cluster the container is running in.
+    'cluster_name': 'instance/attributes/cluster-name'
+}
+
+# Following attributes are derived from environment variables. They are
+# configured via yaml file. For details refer to:
+# https://cloud.google.com/kubernetes-engine/docs/tutorials/
+# custom-metrics-autoscaling#exporting_metrics_from_the_application
+_GKE_ENV_ATTRIBUTES = {
+    # ContainerName is the name of the container.
+    'container_name': 'CONTAINER_NAME',
+
+    # namespace_id is the identifier for the cluster namespace the container
+    # is running in
+    'namespace_id': 'NAMESPACE',
+
+    # pod_id is the identifier for the pod the container is running in.
+    'pod_id': 'HOSTNAME'
+}
+
+# Kubenertes environment variables
+_KUBERNETES_SERVICE_HOST = 'KUBERNETES_SERVICE_HOST'
+
+# inited is used to make sure GCP initialize executes only once.
+inited = False
+
+is_running_on_gcp = False
+
+gcp_metadata_map = {}
+
+
+class GcpMetadataConfig(object):
+    """GcpMetadata represents metadata retrieved from GCP (GKE and GCE)
+    environment. Some attributes are retrieved from the system environment.
+    see : <a href="https://cloud.google.com/compute/docs/
+    storing-retrieving-metadata"> https://cloud.google.com/compute/docs/storing
+    -retrieving-metadata</a>
+    """
+
+    @classmethod
+    def _initialize_metadata_service(cls):
+        """Initialize metadata service once and load gcp metadata into map
+        This method should only be called once.
+        """
+        global inited
```

Comment:
It's a little weird to declare global variable in a class as we aren't using them anywhere outside. Can we declare the variables inside the class?
```
class GcpMetadataConfig(object):
    inited = False
    is_running_on_gcp = False
```